src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java

```diff
@@ -154,13 +163,70 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     }
   }
 
+  /**
+   * Finds a Scan project with the project name from the plugin configuration, or creates such a
+   * "Java" project if one does not exist.
+   *
+   * <p>Note: the Scan API does not expose an endpoint for doing this atomically, so it is possible
+   * that another process creates the project after having determined it to not-exist but before
+   * attempting to create it.
```

Comment:
Is this something the Umbrella team needs to address? Seems like it could cause confusion and I don’t believe we can solve it on our end. Really they need to implement an atomic `findOrCreateProject` endpoint that handles this case I guess.

Reply:
That's right. I will raise it as a concern